app/concepts/reservations/use_cases/create_at_desk.rb

```diff
@@ -0,0 +1,49 @@
+module Reservations
+  module UseCases
+    class CreateAtDesk
+      attr_reader :errors
+
+      def initialize(user:, seance:, seats:)
+        @user = user
+        @seance = seance
+        @seats = seats
+        @errors = []
+      end
+
+      def call
+        return unless seats_selected?
+
+        ActiveRecord::Base.transaction do
+          reservation.tap do |reservation|
+            seats.each do |seat|
+              Ticket.create!(reservation_id: reservation.id, seat: seat)
+            end
+          end
+        end
+        reservation
```

Comment:
```suggestion
        Reservation.create!(
          user: user,
          seance_id: seance.id,
          status: Reservation::CONFIRMED,
          tickets: seats.map { |seat| Ticket.new(seat: seat) }
        )
```

This has two benefits - first, you don't need manually call transaction, as ActiveRecord will handle that now for you. And it should return in two queries - one for creation of reservation and one for creation of all tickets at the same time. Where yours approach creates a query per ticket.

Reply:
Nice, I did not know I could do something like this. What makes it act like a transaction?

Reply:
Because that's a single function call - `Ticket.new` does not yet create ticket in database, it just initialize the object. Then when `Reservation.create!` is called with it will then create both reservation and tickets -  and ActiveRecord will put all of that in transaction by default 😄

Reply:
clever, I like it! :)